src/main/java/io/kestra/plugin/langchain4j/embeddings/MongoDBAtlas.java

```diff
@@ -0,0 +1,132 @@
+import io.kestra.core.models.property.Property;
+import io.kestra.core.runners.RunContext;
+import io.kestra.plugin.langchain4j.domain.EmbeddingStoreProvider;
+import io.swagger.v3.oas.annotations.media.Schema;
+import jakarta.validation.constraints.NotNull;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.experimental.SuperBuilder;
+
+import java.io.IOException;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+@Getter
+@SuperBuilder
+@NoArgsConstructor
+@Plugin(beta = true)
+@JsonDeserialize
+@Schema(
+    title = "MongoDB Atlas Embedding Store"
+)
+public class MongoDBAtlas extends EmbeddingStoreProvider {
+
+    @Schema(title = "The username")
+    private Property<String> username;
+
+    @Schema(title = "The password")
+    private Property<String> password;
+
+    @NotNull
+    @Schema(title = "The host")
+    private Property<String> host;
+
+    @Schema(title = "The database")
+    private Property<String> database;
+
+    @Schema(title = "The connection string options")
+    private Property<Map<String, Object>> options;
+
+    @NotNull
+    @Schema(title = "The collection name")
+    private Property<String> collectionName;
+
+    @NotNull
+    @Schema(title = "The index name")
+    private Property<String> indexName;
+
+    @Schema(title = "The metadata field names")
+    private Property<List<String>> metadataFieldNames;
+
+    @Schema(title = "Create the index")
+    private Property<Boolean> createIndex;
+
+    @Override
+    public EmbeddingStore<TextSegment> embeddingStore(RunContext runContext, int dimension, boolean drop) throws IOException, IllegalVariableEvaluationException {
+
+        var mongoClient = MongoClients.create(buildUri(runContext));
+
+        var renderedCreateIndex = runContext.render(createIndex).as(Boolean.class).orElse(false);
+        var store = MongoDbEmbeddingStore.builder()
+            .fromClient(mongoClient)
+            .databaseName(runContext.render(database).as(String.class).orElseThrow())
+            .collectionName(runContext.render(collectionName).as(String.class).orElseThrow())
+            .indexName(runContext.render(indexName).as(String.class).orElseThrow())
+            .createIndex(renderedCreateIndex)
+            .indexMapping(
+                metadataFieldNames != null ?
+                    IndexMapping.builder()
+                        .dimension(dimension)
+                        .metadataFieldNames(new HashSet<>(runContext.render(metadataFieldNames).asList(String.class)))
+                        .build() : null
+            )
+            .build();
+
+        if (renderedCreateIndex) {
+            // Creating a vector search index can take up to a minute, so this delay allows the index to become queryable
+            try {
+                Thread.sleep(15000);
+            } catch (InterruptedException e) {
+                throw new RuntimeException(e);
+            }
+        }
```

Comment:
This is really not a good idea.
Moreover, if the index already exist it would not take a lot of time.

The best would be to use awaitability and wait for the index to be ready by querying it or any other sort of query that make sure the index is ready.